R/pkg/R/functions.R

```diff
@@ -1752,15 +1752,15 @@ setMethod("toRadians",
 
 #' to_date
 #'
-#' Converts the column into a DateType. You may optionally specify a format
+#' Converts the column into a date column. You may optionally specify a format
 #' according to the rules in:
 #' \url{http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html}.
 #' If the string cannot be parsed according to the specified format (or default),
 #' the value of the column will be null.
-#' The default format is 'yyyy-MM-dd'.
+#' By default, it follows casting rules to a date if the format is omitted.
```

Comment:
It looks the default format is ...

>    * `yyyy`,
>    * `yyyy-[m]m`
>    * `yyyy-[m]m-[d]d`
>    * `yyyy-[m]m-[d]d `
>    * `yyyy-[m]m-[d]d *`
>    * `yyyy-[m]m-[d]dT*`

https://github.com/apache/spark/blob/22691556e5f0dfbac81b8cc9ca0a67c70c1711ca/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L430-L435

which looks used in the casting rule to a date type.